Source/Pages/Gallery/LibraryMediaManager.swift

```diff
@@ -279,15 +286,20 @@ open class LibraryMediaManager {
                                     callback(nil)
                                 }
                             case .failed:
-                                if compressionTypeOverride == nil, let self = self {
-                                    let compressionOverride = presetName == AVAssetExportPresetPassthrough ? YPConfig.video.compression : AVAssetExportPresetPassthrough
+                                if let self = self {
+                                    var retryCount = processingFailedRetryCount
+                                    retryCount += 1
+                                    // Try one more time to process with the export settings on the YPConfig.
+                                    let compressionOverride = YPConfig.video.compression
                                     ypLog("LibraryMediaManager -> Export of the video failed. Reason: \(String(describing: session.error))\n--- Retrying with compression type \(compressionOverride)")
-                                    self.stopExportTimer(for: session)
-                                    self.fetchVideoUrlAndCrop(for: videoAsset, cropRect: cropRect, timeRange: timeRange, shouldMute: shouldMute, compressionTypeOverride: compressionOverride, callback: callback)
-                                }
-                                else {
-                                    ypLog("LibraryMediaManager -> Export of the video failed. Reason: \(String(describing: session.error))")
-                                    self?.stopExportTimer(for: session)
+                                    if retryCount > 1 {
+                                        self.stopExportTimer(for: session)
+                                        callback(nil)
+                                    } else {
+                                        self.stopExportTimer(for: session)
+                                        self.fetchVideoUrlAndCrop(for: videoAsset, cropRect: cropRect, timeRange: timeRange, shouldMute: shouldMute, compressionTypeOverride: compressionOverride, processingFailedRetryCount: retryCount , callback: callback)
+                                    }
+                                } else {
```

Comment:
We should probably also stop the export timer in this else case. When does this actually happen? If we're deallocated?